### PR TITLE
Add actor demo

### PR DESCRIPTION
**Pull request by eiriktsarpalis**

### CODE REVIEW

File: src/Demos/13-actor-example.fsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+type ReplyChannel<'T> = ISendPort<'T>
+
+/// Cloud actor instance
+type CloudActor<'T> =
+    {
+        Sender : ISendPort<'T>
+        Task : ICloudTask<unit>
+        CancellationTokenSource : ICloudCancellationTokenSource
+    }
+with
```
Comment (dsyme) [inline src/Demos/13-actor-example.fsx:34]: I recommend removing "with" in F# code, just align the members correctly (as you have already done)

**Comment by dsyme** [comment]
I'm not sure this is sufficiently high-perf to add to the core Brisk example set just as yet - at least, would be good to set expectations of how many msg/second are expected to be sent/processed in the comments.  I'd also be interested to know what the bottle necks are for this - the channel implementation?

See also this bug which I got on a rewritten version of this (not using a record type) may indicate an issue https://github.com/mbraceproject/MBrace.Azure/issues/23

**Comment by eiriktsarpalis** [comment]
This is definitely not an implementation intended for performance, or serious use for that matter. The `.PostWithReply` implementation is particularly rotten, since it will create a new queue in the service bus every time it is being called. Fault tolerance is also problematic here, since state is kept in-memory at the executing worker. My intention with this demo is to showcase the capabilities of the new `Task` and `ICloudCancellationToken` abstractions when combined with the other primitives. Perhaps I should add a disclaimer to all this at the top.

It would be interesting to investigate the possibility of a practical evolution of this idea. One that actually comes with fault tolerance. This could involve persisting state to store, perhaps even adding some kind of callback support to Channels.

**Comment by dsyme** [comment]
@eiriktsarpalis  - Yes, I see.  In that case I guess it feels like this example is not quite ready yet. I reckon the examples in the Brisk starter pack need to either teach fundamentals (e.g. a tutorial on channels would make sense) or be performance-per-dollar-competitive examples of how to do practical things. I'm not sure all the existing examples quite fit that (including the ones I added :) ) -  if so we should look to improve those too :)

In particular the "actor message throughput" topic is too sensitive: you don't want distributed actor model performance comparisons being made unless you know you're getting up towards the same level of perf as Akka etc.

**Comment by isaacabraham** [comment]
What's the feeling then for this - add it or leave it out?

**Pull request closed without merging**